### PR TITLE
Support spanmetrics connector by default

### PR DESCRIPTION
**Pull request by albertteoh**
## Which problem is this PR solving?
- Supports the new spanmetrics connector by default.
- Some local developer improvements.

## Description of the changes
- Defaults the `PROMETHEUS_QUERY_SUPPORT_SPANMETRICS_CONNECTOR` parameter to `true`.
- The next release will remove this configuration and not support the spanmetrics processor.
- Removes the use of the deprecated jaeger exporter in OTEL config. For more details, please refer to: https://github.com/open-telemetry/opentelemetry-specification/pull/2858.
- Simplifies the Makefile by:
  - Providing defaults in the docker-compose.yml file to prefer the spanmetrics connector.
  - Removing the `run` target in favour of `docker compose up`.
  - Removing the `run-` prefix of the remaining targets.

## How was this change tested?
- Tested the following commands to confirm metrics are visible in the Monitor tab:

```
$ make build
$ make dev
$ make dev-processor
$ docker compose up

# No longer works
$ make run
```

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
~- [] I have added unit tests for the new functionality~
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/metrics/prometheus/options.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func NewOptions(primaryNamespace string) *Options {
 		ServerURL:      defaultServerURL,
 		ConnectTimeout: defaultConnectTimeout,
 
-		SupportSpanmetricsConnector: false,
+		SupportSpanmetricsConnector: defaultSupportSpanmetricsConnector,
```
Comment (albertteoh) [inline plugin/metrics/prometheus/options.go:67]: This is the main (breaking) change. 

The other changes relate to local developer improvements with a similar theme of defaulting to the spanmetrics connector as well.

If we feel those changes should be decoupled into separate PRs, I'm happy to break them up.

File: docker-compose/monitor/otel-collector-config-connector.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ service:
     traces:
       receivers: [otlp, jaeger]
       processors: [batch]
-      exporters: [spanmetrics, jaeger]
```
Comment (albertteoh) [inline docker-compose/monitor/otel-collector-config-connector.yml:32]: [jaeger exporter is deprecated](https://github.com/open-telemetry/opentelemetry-specification/pull/2858), hence the switch to otlp exporter.

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ next release (yyyy-mm-dd)
 #### ⛔ Breaking Changes
 
 * Make OTLP the default exporter in HotROD ([@yurishkuro](https://github.com/yurishkuro) in [#4698](https://github.com/jaegertracing/jaeger/pull/4698))
+* Support spanmetrics connector by default ([@albertteoh](https://github.com/albertteoh) in [#4704](https://github.com/jaegertracing/jaeger/pull/4704))
```
Comment (yurishkuro) [inline CHANGELOG.md:29]:
```suggestion
* [SPM] Support spanmetrics connector by default ([@albertteoh](https://github.com/albertteoh) in [#4704](https://github.com/jaegertracing/jaeger/pull/4704))
```

**Review by yurishkuro — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]
